regression/rt_ar_react_1x2v_quadraticV.c

```diff
@@ -0,0 +1,969 @@
+    .chargeElc = qe, 
+    .massElc = me, 
+    .chargeIon = qi, 
+    .massIon = mi,
+    .massAr = mAr,
+    .Te = Te, 
+    .Ti = Ti,
+    .TAr = TAr,
+    .c_s = c_s, 
+    .nuElc = nuElc, 
+    .nuIon = nuIon, 
+    .nuFrac = nuFrac, 
+    .B0 = B0, 
+    .n0 = n0, 
+    .n0Ar = n0Ar,
+    .n0Ar1 = n0Ar1,
+    .n0Ar2 = n0Ar2,
+    .n0Ar3 = n0Ar3,
+    .n0Ar4 = n0Ar4,
+    .n0Ar5 = n0Ar5,
+    .n0Ar6 = n0Ar6,
+    .n0Ar7 = n0Ar7,
+    .n0Ar8 = n0Ar8,
+    .n0Ar9 = n0Ar9,
+    .n0Ar10 = n0Ar10,
+    .n0Ar11 = n0Ar11,
+    .n0Ar12 = n0Ar12,
+    .n0Ar13 = n0Ar13,
+    .n0Ar14 = n0Ar14,
+    .n0Ar15 = n0Ar15,
+    .n0Ar16 = n0Ar16,
+    .n0Ar17 = n0Ar17,
+    .n0Ar18 = n0Ar18,
+    .Lz = Lz, 
+    .kperp = kperp, 
+    .n_src = n_src,
+    .T_src = T_src,
+    .vpar_max_elc = vpar_max_elc, 
+    .mu_max_elc = mu_max_elc, 
+    .vpar_max_ion = vpar_max_ion, 
+    .mu_max_ion = mu_max_ion,
+    .vpar_max_Ar = vpar_max_Ar, 
+    .mu_max_Ar = mu_max_Ar, 
+    .finalTime = finalTime, 
+    .numFrames = numFrames,
+  };
+  return ctx;
+}
+
+struct all_of_1species_reactions {
+  struct gkyl_gyrokinetic_react_type recombination[GKYL_MAX_SPECIES];
+  struct gkyl_gyrokinetic_react_type ionization[GKYL_MAX_SPECIES];
+};
+
+struct all_of_1species_reactions*
+create_isonuclear_react(double ion_mass, double electron_mass, int max_z, enum gkyl_ion_type ion_id, char* electron_name, char* ion_name)
+{
+  static struct all_of_1species_reactions reactions[4];  // index = gkyl_react_self_type
+  // 0 = elc, 1=ion, 2 = donor, 3 = recvr
+  char ion_nm[128];
+  char donor_nm[128];  // Same as recvr_nm
+  for (int j=0; j<4; j++) {
+    for (int i=0; i<max_z; i++) {
+      snprintf(ion_nm, sizeof(ion_nm), "%s%d", ion_name, i+1);
+      snprintf(donor_nm, sizeof(donor_nm), "%s%d", ion_name, i);
+      reactions[j].recombination[i].react_id = GKYL_REACT_RECOMB;
+      reactions[j].recombination[i].ion_id = ion_id;
+      strcpy(reactions[j].recombination[i].elc_nm, electron_name);
+      reactions[j].recombination[i].charge_state = i;
+      reactions[j].recombination[i].ion_mass = ion_mass;
+      reactions[j].recombination[i].elc_mass = electron_mass;
+      strcpy(reactions[j].recombination[i].ion_nm, ion_nm);
+      strcpy(reactions[j].recombination[i].recvr_nm, donor_nm);
+      reactions[j].recombination[i].type_self = j;
+      
+      reactions[j].ionization[i].react_id = GKYL_REACT_IZ;
+      reactions[j].ionization[i].ion_id = ion_id;
+      strcpy(reactions[j].ionization[i].elc_nm, electron_name);
+      reactions[j].ionization[i].charge_state = i;
+      reactions[j].ionization[i].ion_mass = ion_mass;
+      reactions[j].ionization[i].elc_mass = electron_mass;
+      strcpy(reactions[j].ionization[i].ion_nm, ion_nm);
+      strcpy(reactions[j].ionization[i].donor_nm, donor_nm);
+      reactions[j].ionization[i].type_self = j;
+    }
+  }
+  return reactions;
+}
+
+void
+write_data(struct gkyl_tm_trigger *iot, gkyl_gyrokinetic_app *app, double tcurr)
```

Comment:
@jRoeltgen please stick to the structure and workflow of the 2 I/O functions in other regression tests